ambry-store/src/main/java/com.github.ambry.store/BlobStore.java

```diff
@@ -216,6 +249,29 @@ private MessageWriteSetStateInStore checkWriteSetStateInStore(MessageWriteSet me
     }
   }
 
+  /**
+   * Checks the used capacity of the store against the configured percentage thresholds to see if the store
+   * should be read-only or read-write
+   * @param totalCapacity total capacity of the store in bytes
+   * @param usedCapacity total used capacity of the store in bytes
+   */
+  private void checkCapacityAndUpdateClusterManager(long totalCapacity, long usedCapacity) {
```

Comment:
And I think this should actually be named `checkCapacityAndUpdateClusterParticipant`. 

`BlobStore` does not really deal with `ClusterManager` today really, mainly because the store for all purposes is a "local" entity that does not know that it is part of a cluster. To that effect, maybe the right thing to do is to have a `WriteStatusDelegate` interface that the store deals with and `ClusterParticipantWriteStatusDelegate` would be an implementation of it that is passed down to the store. Might be an overkill, but would encourage @dharju and @vgkholla to discuss this.

Reply:
@vgkholla and I chatted and we think that it would be fine to rename `ClusterParticipantWriteStatusDelegate` to `WriteStatusDelegate` instead of making a whole interface since we don't really see how in the future we're going to be adding new implementations for a hypothetical `WriteStatusDelegate`

Reply:
Makes sense.